FogCreek/Glitch-Community
# Sunset stitch: Super User Banner and Toggle Button

## Links
* https://sunset-stitch.glitch.me
* https://glitch.manuscript.com/f/cases/3326596/
* https://glitch.manuscript.com/f/cases/3327754/
* https://glitch.com/edit/#!/support-toggle

## GIF/Screenshots:
![Screen Shot 2019-04-03 at 1 53 55 PM](https://user-images.githubusercontent.com/6620164/55501241-fff51880-5617-11e9-9cb0-d31171c9cf5d.png)
![Screen Shot 2019-04-03 at 1 54 04 PM](https://user-images.githubusercontent.com/6620164/55501242-fff51880-5617-11e9-8970-0bd02fbe2fb5.png)

## Changes:
* normal users
    * should have no impact
* users invited to the supah secret project (ask cori)
    * all super users should see a big red banner with a button to turn off their super user status
    * to see the banner all the time users can run `window.localStorage.setItem('showSupportBanner', true)` and they will see the banner all the time until they clear their localStorage
    * when they have their super user feature turned off, they are able to hide the banner with a button click (which clears that part of localStorage)

## How To Test:
* log in as a user with the super_user feature, verify you see the banner
* or if you're not a super user such as myself you can mock it by adding the following code to line 74 on router.js:
`currentUser.features = [{ name: "super_user", expiresAt: "2019-11-01T14:11:01.233Z" }] `
* OR you can also put `window.localStorage.setItem("showSupportBanner", true)` in your console to see the banner

## Feedback I'm looking for:
* would love some super users to verify that they see the banner and that this meets their needs
* make sure this isn't too funky code wise. I did use inline styles just because it seemed easy/simple but happy to make this a real component or use css if we think that's best! 

## Things left to do before deploying:
- [x] fix CORS issue
- [x] get a super_user to qa it
- [x] verify I can't enable super user status if I'm not a member of the secret project
- [x] verify once more this doesn't do anything funky on other browsers for some reason.
